students/arkadiusz_kasprzyk/lesson_06_dicts/uppercase.py

```diff
@@ -0,0 +1,91 @@
+# uppercase.py
+
+
+def capitalize_word(word):
+    """
+    Parameters
+    ----------
+    word: str
+
+    Returns
+    -------
+    word_cap: str
+        First letter of `word` turned to uppercase (if it's a letter).
+
+    Examples
+    --------
+    capitalize_word("spam")         # 'Spam'
+    capitalize_word("Spam")         # 'Spam'
+    capitalize_word("_spam")        # '_spam'
+    capitalize_word(" spam")        # ' spam'
+    capitalize_word("_spam eggs")   # '_spam eggs'
+    """
+    word0 = word[0]
+    char0 = ord(word0)
+    if char0 in range(97, 123):
+        word0 = chr(char0 - 32)
+
+    word_cap = word0 + word[1:]
+    return word_cap
+
+
+def capitalize(text):
+    """
+    Parameters
+    ----------
+    text: str
+
+    Returns
+    -------
+    text_cap: str
+        Each word of `txt` capitalized.
+
+    Examples
+    --------
+    capitalize("spam and eggs")     # 'Spam And Eggs'
+    capitalize(" spam and eggs")    # ' Spam And Eggs'
+    capitalize("spam and eggs ")    # 'Spam And Eggs '
+    capitalize("?sPam aNd egGs ")   # '?sPam ANd EgGs '
+    capitalize("  spam  .eGgs . ")  # '  Spam  .eGgs . '
+    """
+    text_cap = ""
+    a = 0
+    n = len(text)
+    while a < n:
```

Comment:
If you used `split()` and `join()`, it would be easier to read.

Reply:
I'm not sure as there is a problem of remembering size (number) of spaces between words. See the last 2 examples (I have just added the last). Does split() return the separators list in case of default sep?

Reply:
If you want to retain spaces, then `split` won't work with default sep (handling newlines, multiple spaces and tabs), but it will work just as good with space only.